### PR TITLE
[core] Add simple fuzz test for storage2::BitStash

### PR DESCRIPTION
**Pull request by cmichi**
Can be run via `cd core/ && cargo test --features ink-fuzz-tests -- bitstash::fuzz_tests`.

### CODE REVIEW

File: core/src/storage2/collections/bitstash/fuzz_tests.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use super::BitStash;
+use crate::env;
+
+/// Conducts repeated insert and remove operations into the stash by iterating
+/// over `xs`. Typically the `xs` and `inserts_each` arguments are provided
+/// by our fuzzing engine in an iterative manner.
+///
+/// For each odd `x` in `xs` a number of put operations are executed.
+/// For each even `x` it is asserted that the previously inserted elements
+/// are in the stash and they are taken out subsequently.
```
Comment (Robbepop) [inline core/src/storage2/collections/bitstash/fuzz_tests.rs:24]: Btw.: Just curious. Is this kind of pattern common in fuzz tests?

Comment (cmichi) [reply core/src/storage2/collections/bitstash/fuzz_tests.rs:24]: Nope, I haven't seen it elsewhere. I started this pattern while implementing `BTreeMap` and found a couple of corner cases. The idea is to maximize randomness for insert/remove operations. Yeah so it has worked well so far, but of course I'm open to other ideas!

Comment (Robbepop) [reply core/src/storage2/collections/bitstash/fuzz_tests.rs:24]: Would be very nice if you could investigate this further in how other crates that are doing fuzz tests are using their fuzzers to randomize their inputs.
